packages/client/src/views/Performance/RegistrationRates.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as React from 'react'
+import { RouteComponentProps } from 'react-router'
+
+import { PerformanceContentWrapper } from '@client/views/Performance/PerformanceContentWrapper'
+import {
+  TertiaryButton,
+  ICON_ALIGNMENT
+} from '@opencrvs/components/lib/buttons'
+import { WrappedComponentProps, injectIntl } from 'react-intl'
+import { buttonMessages } from '@client/i18n/messages'
+import { ArrowBack } from '@opencrvs/components/lib/icons'
+import styled from '@client/styledComponents'
+import { Query } from '@client/components/Query'
+import { GET_CHILD_LOCATIONS_BY_PARENT_ID } from './queries'
+import { messages } from '@client/i18n/messages/views/performance'
+import { PerformanceSelect } from './PerformanceSelect'
+import { connect } from 'react-redux'
+import { transformChildLocations } from '@client/views/Performance/utils'
+import { goToOperationalReport } from '@client/navigation'
+
+const { useState } = React
+const Header = styled.h1`
+  color: ${({ theme }) => theme.colors.copy};
+  ${({ theme }) => theme.fonts.h2Style};
+`
+const NavigationActionContainer = styled.div`
+  margin-top: 24px;
+`
+const ActionContainer = styled.div`
+  display: flex;
+  & > button {
+    margin-right: 8px;
+  }
+
+  & > :last-child {
+    margin-right: 0;
+  }
+`
+enum REG_RATE_BASE {
+  TIME = 'TIME',
+  LOCATION = 'LOCATION'
+}
+
+interface IDispatchProps {
+  goToOperationalReport: typeof goToOperationalReport
+}
+type IRegistrationRateProps = RouteComponentProps<{ eventType: string }> &
+  WrappedComponentProps &
+  IDispatchProps
+
+function RegistrationRatesComponent(props: IRegistrationRateProps) {
+  const [base, setBase] = useState<REG_RATE_BASE>(REG_RATE_BASE.TIME)
+
+  const {
+    intl,
+    history: {
+      location: { state }
+    },
+    match: {
+      params: { eventType }
+    },
+    goToOperationalReport
+  } = props
+  const { title, selectedLocation } = state
+
+  return (
+    <PerformanceContentWrapper hideTopBar>
+      <NavigationActionContainer>
+        <TertiaryButton
+          id="reg-rates-action-back"
+          icon={() => <ArrowBack />}
+          align={ICON_ALIGNMENT.LEFT}
+          onClick={() => goToOperationalReport(selectedLocation)}
+        >
+          {intl.formatMessage(buttonMessages.back)}
+        </TertiaryButton>
+      </NavigationActionContainer>
+      <Header id="reg-rates-header">{title}</Header>
+
+      <Query
+        query={GET_CHILD_LOCATIONS_BY_PARENT_ID}
```

Comment:
Why do we need to query child locations here ?

Reply:
The issue required to add option "By Unions" if the selected location has child locations. So, in order to get child locations I had to add this query.
 
![image](https://user-images.githubusercontent.com/42269993/79842466-ae622c00-83da-11ea-9dd7-029bc6726e6d.png)

Reply:
Fetching the full list of child locations over the network in-order to decide whether current location has any child or not, feels a bit expensive to me. Can't we introduce a new query on gateway to determine whether a location has any child or not. ex: `hasChildLocation`

Reply:
Think about doing a query from national level, it will fetch the full list of locations in a country just to decide whether it has any child or not.

Reply:
I agree. In that case, we might have to send the `jurisdictionType` of the child locations too...

Reply:
Thanks, @mushrafulhoque-dsi, I am pushing the changes in my next commit.